guacamole-ext/src/main/java/org/apache/guacamole/net/auth/AbstractAuthenticatedUser.java

```diff
@@ -41,4 +42,12 @@ public void invalidate() {
         // Nothing to invalidate
     }
 
+    public Map<String, String> getAttributes() {
```

Comment:
So, I think there are two options here:

1. Don't implement these, here, since `AbstractAuthenticatedUser` implements `AuthenticatedUser`, which implements `Attributes`, these should already be defined at that level.  If you don't implement them, here, any classes that extend `AbstractAuthenticatedUser` will need to override them.
2. Implement them, here, so that there's a base implementation, but use the `@Override` tag so that it's understood those are overriding methods from another class (`AuthenticatedUser`).  This means you won't have to implement them in every single class that extends `AbstractAuthenticatedUser` except where you want to change this behavior.

Option 2 is definitely simpler, or at least requires fewer changes to other extensions, so that's probably the easiest to go with for the time being - that just means adding the `@Override` tag for this method and `setAttributes()`.

Reply:
I think Option 2 is the way to go. Personally, I don't think we should mess so much with the other extensions.